test/ecto/dev_logger_test.exs

```diff
@@ -146,7 +147,36 @@ defmodule Ecto.DevLoggerTest do
     post = Repo.get!(Post, post_id)
     post = post |> Ecto.Changeset.change(string: nil) |> Repo.update!()
     Repo.delete!(post)
+  end
+
+  describe "generates correct sql for structs" do
+    test "ip field" do
+      ip = %Postgrex.INET{address: {127, 0, 0, 1}, netmask: 24}
+
+      log = capture_log(fn -> Repo.insert!(%Post{ip: ip}) end)
+
+      assert strip_ansi(log) =~
+               ~S|INSERT INTO "posts" ("ip") VALUES ('127.0.0.1/24') RETURNING "id"|
+    end
+
+    test "macaddr field" do
+      macaddr = %Postgrex.MACADDR{address: {8, 1, 43, 5, 7, 9}}
+
+      log = capture_log(fn -> Repo.insert!(%Post{macaddr: macaddr}) end)
+
+      assert strip_ansi(log) =~
+               ~S|INSERT INTO "posts" ("macaddr") VALUES ('08:01:2B:05:07:09') RETURNING "id"|
+    end
 
+    test "enum field" do
+      log = capture_log(fn -> Repo.insert!(%Post{enum: [:foo, :baz]}) end)
+
+      assert strip_ansi(log) =~
+               ~S|INSERT INTO "posts" ("enum") VALUES ({1,5}/*foo,baz*/) RETURNING "id"|
```

Comment:
oh, `enum` field represents an array.
Could you rename `enum` field to `array_of_enums` and add `enum` field to test a single value?

Reply:
I refactored it so the struct keeps only a single value. I can insert the array to database